src/Task/LockDeploymentTask.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+
+namespace TYPO3\Surf\Task;
+
+/*
+ * This file is part of TYPO3 Surf.
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+use TYPO3\Surf\Domain\Model\Application;
+use TYPO3\Surf\Domain\Model\Deployment;
+use TYPO3\Surf\Domain\Model\Node;
+use TYPO3\Surf\Domain\Model\Task;
+use TYPO3\Surf\Domain\Service\ShellCommandServiceAwareInterface;
+use TYPO3\Surf\Domain\Service\ShellCommandServiceAwareTrait;
+use TYPO3\Surf\Exception\DeploymentLockedException;
+
+/**
+ * Lock deployment task
+ */
+final class LockDeploymentTask extends Task implements ShellCommandServiceAwareInterface
+{
+    const LOCK_FILE_NAME = 'deploy.lock';
+
+    use ShellCommandServiceAwareTrait;
+
+    /**
+     * @param Node $node
+     * @param Application $application
+     * @param Deployment $deployment
+     * @param array $options
+     */
+    public function execute(Node $node, Application $application, Deployment $deployment, array $options = [])
+    {
+        $deploymentLockFile = sprintf('%s/.surf/%s', escapeshellarg($application->getDeploymentPath()), self::LOCK_FILE_NAME);
+        $locked = $this->shell->execute(sprintf('test -f %s', $deploymentLockFile), $node, $deployment, true);
+        if ($locked) {
```

Comment:
@schreiberten This is the problem: if the lock file exists, `test` exit with `0` and `$this->shell->execute` returns the [output](https://github.com/TYPO3/Surf/blob/master/src/Domain/Service/ShellCommandService.php#L44) of the command which is empty. An empty string evaluates to `false` without type safe comparison.
If the lock file doesn't exists test the exit code of `test` is 1 so `$this->shell->execute` returns false.

Now we could simply do a type safe comparison or we change the executed command to something that returns 0 if file doesn't exists or 1 if it exists, like `$this->shell->execute(sprintf('if [ -f %s ]; then echo 1; else echo 0; fi', $deploymentLockFile), $node, $deployment);`

What do you think is better to read and understand?

Reply:
Or we could add a ShellCommandService parameter to just return true or false.

Reply:
@t3easy Awesome. I think your example "$this->shell->execute(sprintf('if [ -f %s ]; then echo 1; else echo 0; fi', $deploymentLockFile), $node, $deployment);" looks fine for me. 
Would like to see this feature merged asap.

Reply:
OK, I'll prepare that, but as we're not testing the deployment ATM, give me some time to build an docker image and test the deployment.

Reply:
looks good so far